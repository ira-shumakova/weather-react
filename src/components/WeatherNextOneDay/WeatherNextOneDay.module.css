.oneDay {
  background-color: #44ba8f70;
  backdrop-filter: blur(20px);
  border: 2px solid #06201396;
  border-radius: 20px;
  margin-top: 1rem;
  height: 6.25rem;
}

.dailyStyle {
  text-align: center;
}

.dailyTemp {
  margin-top: 0px;

}

.dailyTemp {
  font-weight: 800;
  font-size: 1.25rem;
  margin-bottom: 0px;
  margin-top: 0.5rem;
}

.minTemp {
  font-weight: 400;
  font-size: 1rem;
  color: #292627;
  margin-top: 0px;
  padding-left: 1rem;
}

.day {
  margin: 0px;
}