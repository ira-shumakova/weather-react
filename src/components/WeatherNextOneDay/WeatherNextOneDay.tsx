import moment from "moment";
import { IDaysWeather } from "../../dto/types";
import classes from './WeatherNextOneDay.module.css';
import iconLink from "../../utils/getIcons";
import { Grid } from "@mui/material";

type Props = {
  day: IDaysWeather,
}

const WeatherNextOneDay = ({day} : Props) => {
  const convertDay = (time: number) => {
    return moment.unix(time).local().format('dddd');
  }
  return (
    <div 
      className={classes.oneDay}
      data-bs-toggle="tooltip"
      data-bs-placement="left"
      title={day.weather[0].description}
    >
      <Grid 
        container spacing={0}
        justifyContent="space-around"
        className={classes.dailyStyle}
      >
        <Grid item xs={12}>
          <h3 className={classes.day}>
            {convertDay(day.dt)}
          </h3>
        </Grid>
        <Grid item xs={4} md={6} xl={7}>
          <img 
            src={iconLink(day.weather[0].icon)} 
            alt={day.weather[0].description} 
            />
        </Grid>
        <Grid item xs={8} md={6} xl={5}>
          <h3 className={classes.dailyTemp}>
            {Math.round(day.temp.max)}&deg;C
          </h3>
          <p className={classes.minTemp}>
            {Math.round(day.temp.min)}&deg;C
          </p>
        </Grid>
      </Grid>
      
    </div>
  )
}

export default WeatherNextOneDay;