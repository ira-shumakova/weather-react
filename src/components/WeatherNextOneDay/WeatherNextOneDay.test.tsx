import { render } from "@testing-library/react";
import WeatherNextOneDay from "./WeatherNextOneDay";

describe("WeatherNextOneDay", ()=> {
  test("matches snapshot", () => {
    const day = {
      temp: {
        max: 14,
        min: 6,
      },
      feels_like: 13,
      humidity: 55,
      pressure: 1017,
      sunrise: 1681960920,
      sunset: 1682011401,
      wind_speed: 6.68,
      dt: 1681984800,
      weather: [
        {
          description: "broken clouds",
          icon: "04d",
          main: "Clouds"
        },
      ],
    };
    const {container} = render(<WeatherNextOneDay day={day} />)
    expect(container).toMatchSnapshot();
  });
});