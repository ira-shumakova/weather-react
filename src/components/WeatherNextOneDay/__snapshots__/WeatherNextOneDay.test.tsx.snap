// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherNextOneDay matches snapshot 1`] = `
<div>
  <div
    class="oneDay"
    data-bs-placement="left"
    data-bs-toggle="tooltip"
    title="broken clouds"
  >
    <div
      class="MuiGrid-root MuiGrid-container dailyStyle css-11j0bs5-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h3
          class="day"
        >
          Thursday
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-6 MuiGrid-grid-xl-7 css-adoetp-MuiGrid-root"
      >
        <img
          alt="broken clouds"
          src="04d.svg"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-6 MuiGrid-grid-xl-5 css-rsm9di-MuiGrid-root"
      >
        <h3
          class="dailyTemp"
        >
          14
          °C
        </h3>
        <p
          class="minTemp"
        >
          6
          °C
        </p>
      </div>
    </div>
  </div>
</div>
`;
